python/ray/ml/utils/datasets.py

```diff
@@ -0,0 +1,59 @@
+from typing import Optional, Tuple, Union
+from ray.data import Dataset
+
+
+def train_test_split(
+    dataset: Dataset,
+    test_size: Union[int, float],
+    *,
+    shuffle: bool = False,
+    seed: Optional[int] = None,
+) -> Tuple[Dataset, Dataset]:
+    """Split a Dataset into train and test subsets.
+
+    Example:
+        .. code-block:: python
+
+            import ray
+            from ray.ml import train_test_split
+
+            ds = ray.data.range(8)
+            train, test = train_test_split(ds, test_size=0.25)
+            print(train.take())  # [0, 1, 2, 3, 4, 5]
+            print(test.take())  # [6, 7]
+
+    Args:
+        dataset: Dataset to split.
+        test_size: If float, should be between 0.0 and 1.0 and represent the proportion
+            of the dataset to include in the test split. If int, represents the
+            absolute number of test samples.
```

Comment:
Could you add a note that the train split will always be the complement of the test split? I know that it should be obvious, but shouldn't hurt for those that would like to see that explicitly called out.